### PR TITLE
completes testing for /api endpoint

### PR DESCRIPTION
**Pull request by kimmimonkey**
Hi NCers, 

I've completed logic and testing for task 3. I couldn't see how to do it by requiring in the file when I couldn't export it from the file itself, so I'd love to hear your thoughts!

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -69,5 +54,30 @@ describe("GET /api/topics", () => {
...
     })
-
 })
+
+describe("GET /api", () => {
+    test("status: 200", () => { 
+        return request(app)
+        .get("/api")
+        .expect(200)
+
```
Comment (Harrison365) [inline __tests__/app.test.js:64]: This assertion can be done in the other test. No need for it to be separate :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -69,5 +54,30 @@ describe("GET /api/topics", () => {
...
+        })
+    })
+    test("responds with an endpoints object, where each endpoint has a description property", () => {
+        return request(app)
+        .get("/api")
+        .then( async ({body: {endpoints}}) => {
+            const data = await fs.readFile("./endpoints.json")
+            const parsedData = JSON.parse(data)
+            expect(endpoints).toEqual(parsedData)
+        })
```
Comment (Harrison365) [inline __tests__/app.test.js:80]: Could have used require here instead, which may be easier as its synchronous 👍 Nothing wrong with this though 🙂

File: models/topicsModel.js [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
     return db
     .query(`SELECT * FROM topics;`)
     .then((body) => body.rows); 
 }
 
+exports.fetchAllEndpoints = async() => {
+    const data = await fs.readFile("./endpoints.json");
+    const endpoints = JSON.parse(data)
+    return endpoints;
+}
```
Comment (Harrison365) [inline models/topicsModel.js:14]: All good 👍

**Comment by nc-saleh** [comment]
*TASK* - /api/topics

*FEEDBACK* -  Before iterating over the requested array of objects and writing assertions for each iteration, we should first write an assertion that expects the requested array to be a certain length. The reason for this is that, if by chance, your endpoint returned an empty array, then the callback function inside of your forEach would never be invoked. This would mean that the assertions you have written inside of your forEach would never run and this could lead to a false positive.


The way you have your tests set up, you should be doing the above in both as best practice, not just your second test. However a further point is that I don't think  you need your first test. You first test checks for every element being an object in your array. You second test would fail if this wasn't the case as you are checking the structure of those objects. 

- Also you are missing an assertion for an expected status code which in this case would be a 200. Supertest has a method for this.

**Pull request merged**